### PR TITLE
Feat/post delete feature and fetched latest changes on main

### PR DESCRIPTION
**Pull request by mikco-hipe**
### OVERVIEW

Added destroy() function to PostController
Used SoftDeletes to mark posts with deleted timestamps
Added delete in UserPostPlicy
Added a delete button on show.blade.php
Added destroy on routes

### SNIPPETS
Screenshot 2023-10-09 105548
Screenshot 2023-10-09 110404
image
image

### CODE REVIEW

File: laravel-app/resources/views/home.blade.php [hunk 1]
```diff
@@ -25,6 +25,13 @@
                 <button type="button" class="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
             </div>
         </div>
+
+        @if (session('success'))
+        <div class="alert alert-success">
+            {{ session('success') }}
+        </div>
```
Comment (hipe01mark) [inline laravel-app/resources/views/home.blade.php:32]: indention please!

**Review by hipe01mark — APPROVED** [review]
LGTM 🙇

File: laravel-app/resources/views/home.blade.php (hunk 1 above)
Comment (mikco-hipe) [reply laravel-app/resources/views/home.blade.php:32]: Yes sir 🙇

File: laravel-app/app/Http/Controllers/PostController.php [hunk 2]
```diff
@@ -91,4 +91,14 @@ public function update(UserPostRequest $request, UserPost $post): RedirectRespon
 
         return redirect()->back()->with('error', 'Failed to update post');
     }
+
+    /**
+     * Delete a post
```
Comment (rafael-parayno) [inline laravel-app/app/Http/Controllers/PostController.php:96]: Remove comment already understandable

File: laravel-app/app/Policies/UserPostPolicy.php [hunk 3]
```diff
@@ -14,4 +14,12 @@ public function edit(User $user, UserPost $post): bool
     {
         return $user->id === $post->user_id;
     }
+
+    /**
+     * Determine whether the user can delete the post.
```
Comment (rafael-parayno) [inline laravel-app/app/Policies/UserPostPolicy.php:19]: Remove

**Review by rafael-parayno — APPROVED** [review]

File: laravel-app/app/Http/Controllers/PostController.php (hunk 2 above)
Comment (mikco-hipe) [reply laravel-app/app/Http/Controllers/PostController.php:96]: Got it sir 🙇

File: laravel-app/app/Policies/UserPostPolicy.php (hunk 3 above)
Comment (mikco-hipe) [reply laravel-app/app/Policies/UserPostPolicy.php:19]: Got it sir 🙇

**Review by hipe01mark — APPROVED** [review]
LGTM 🙇

File: laravel-app/resources/views/post/show.blade.php [hunk 4]
```diff
@@ -22,20 +22,38 @@
...
                         @include('post.form_like')
                     </div>
                     <div class="row">
+                        <div class="row text-center">
                         @can('edit', $post)
                             <a href="{{ route('post.edit', $post) }}" class="btn btn-dark">Edit Post</a>
                         @endcan
+
+                        @can('delete', $post)
+                            <form id="delete-post-form" method="POST" action="{{ route('post.destroy', $post) }}">
```
Comment (hipe-francis) [inline laravel-app/resources/views/post/show.blade.php:42]: Shouldn't the destroy param be an id/column value?

**Review by hipe-francis — CHANGES_REQUESTED** [review]

**Review by hipe-francis — APPROVED** [review]
LGTM 🚀

**Pull request merged**